### ISSUE TITLE
Can't always pick the right color in some zones

### ISSUE THREAD

**Issue by redtide** [agateau/colorpick#5]
I noticed that, after changing the hidpi scale factor some Qt applications, including colorpick, doesn't work anymore as expected, grabbing colors/images wrongly. I suspect there is some code missing that deals with the screen scale factor. ATM I noticed only Flameshot not suffering by this issue.

This is quite a blocker for #3.

**Comment by agateau** [comment]
Oh, I don't have an hidpi screen around, so it's tricky for me to reproduce.

I suspect the color picker code needs to take [QScreen::devicePixelRatio()](https://doc.qt.io/qt-5.15/qscreen.html#devicePixelRatio-prop) into account. Do you feel like giving it a try?

**Comment by redtide** [comment]
I'll do!

**Comment by redtide** [comment]
A bit late but I did...
currently I use a scalefactor of 1 and all is fine, except in qtCreator which I set it to 1.5 because its fonts are too tiny with 1.
Running colorpick in the debugger I get a lens drawn of about a 2/3 of its dimension, so it works after a test in the [paintEvent()](https://github.com/agateau/colorpick/blob/master/src/colorpicker.cpp#L86) with:
```cpp
QPixmap pix = mPixmap.scaled(GRAB_SIZE * 1.5 * MAGNIFY,
              GRAB_SIZE * 1.5 * MAGNIFY);
```
but the original issue deals with some strange offset: if I add 5px to x and y in [grabWindow()](https://github.com/agateau/colorpick/blob/master/src/colorpicker.cpp#L144) it "works":
```cpp
mPixmap = screen->grabWindow(wid, pos.x() - GRAB_RADIUS + 5,
          pos.y() - GRAB_RADIUS + 5, GRAB_SIZE, GRAB_SIZE);
```
I guess I should cache the pixel ratio in a `qreal` variable and set it in the constructor since it needs `findScreenAt()`, but I have no idea about those 5 pixels offset. Thoughts?

Unrelated: in a magnifier app I wrote, I set different scales, changing from 2x to 5x when using the [mouse wheel](https://github.com/qtilities/magnifiqus/blob/master/src/mainwindow.cpp#L172); in your opinion, could be useful to set a "variable MAGNIFY", say 4, 8 and 16 or something?
